website/docs/tools/optim.mdx

```diff
@@ -87,10 +87,15 @@ import { constantFolding } from '@publicodes/tools/optims'
 const optimizedRules = constantFolding(
   // A publicode engine instantiated with the rules to optimize.
   new Engine(baseRules),
-  // A predicate returning true if the rule needs to be kept.
-  ([ruleName, ruleNode]) => {
-    return ['root', 'root . bis'].includes(ruleName) ||  ruleNode.rawNode['to keep']
-  }
+  {
+    // A predicate returning true if the rule needs to be kept.
+    toKeep: (rule) => {
+      return (
+        ['root', 'root . bis'].includes(rule.dottedName) ||
+        !!rule.rawNode['to keep']
```

Comment:
Pourquoi utiliser `!!` ici ?

Reply:
´rule.rawNode['to keep']´ est de type ´unknown´ là où ´toKeep´ attend un booléen.

Le !! est là pour la conversion de type (et j'imagine que ´to keep´ peut être ´undefined´ ?).